jigglypuffRL/deeprl/ppo1/ppo1.py

```diff
@@ -0,0 +1,229 @@
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as opt
+from torch.autograd import Variable
+from torch.distributions import Categorical
+import gym
+
+from jigglypuffRL.common import (
+    MlpPolicy,
+    MlpValue,
+    get_policy_from_name,
+    get_value_from_name,
+)
+
+
+class PPO1:
+    """
+    Proximal Policy Optimization algorithm (Clipped policy).
+    Paper: https://arxiv.org/abs/1707.06347
+    :param policy: (str) The policy model to use (MlpPolicy)
+    :param value: (str) The value function model to use (MlpValue)
+    :param env: (Gym environment) The environment to learn from
+    :param timesteps_per_actorbatch: (int) timesteps per actor per update
+    :param gamma: (float) discount factor
+    :param clip_param: (float) clipping parameter epsilon
+    :param actor_batchsize: (int) trajectories per optimizer epoch
+    :param epochs: (int) the optimizer's number of epochs
+    :param lr_policy: (float) policy network learning rate
+    :param lr_value: (float) value network learning rate
+    :param policy_copy_interval: (int) number of optimizer before copying params from new policy to old policy
+    :param tensorboard_log: (str) the log location for tensorboard (if None, no logging)
+    :param seed (int): seed for torch and gym
+    :param device (str): device to use for tensor operations; 'cpu' for cpu and 'cuda' for gpu
+    """
+
+    def __init__(
+        self,
+        policy,
+        value,
+        env,
+        timesteps_per_actorbatch=200,
+        gamma=0.99,
+        clip_param=0.2,
+        actor_batch_size=8,
+        epochs=1000,
+        lr_policy=0.001,
+        lr_value=0.005,
+        policy_copy_interval=20,
+        tensorboard_log=None,
```

Comment:
Why is default value of ```tensorboard_log=None```? As in, what else can it be other than ```SummaryWriter```?
Maybe it would make more sense to keep it to a boolean value? Depending on whether the user wants tb logging or not

Reply:
`tensorboard_log` expects a path (for log dir) as of now. If nothing is provided, logging won't be done. If we set it to a boolean, we will need another parameter for logging location right?

Reply:
Oh ok. I think we will need a logger though, soon. There we can specify the options similar to how stable baselines does it. We can just pass the logger instead of the main class

Reply:
Will add this to issues